pkg/controller/networkpolicy/validate.go

```diff
@@ -20,13 +20,40 @@ import (
 	"strconv"
 
 	admv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog"
 
 	secv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
 )
 
+// createValidateFn is a function signature which should be implemented by any
+// function that validates a CREATE request for a resource.
+type createValidateFn func(curObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
+
+// updateValidateFn is a function signature which should be implemented by any
+// function that validates an UPDATE request for a resource.
+type updateValidateFn func(curObj, oldObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
+
+// deleteValidateFn is a function signature which should be implemented by any
+// function that validates a DELETE request for a resource.
+type deleteValidateFn func(oldObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
```

Comment:
Can we use an interface for this, I think it is more idiomatic and it will end up being less verbose IMO

```go
type Validator interface {
    ValidateCreate(curObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
    ValidateModify(curObj, oldObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
    ValidateDelete(oldObj interface{}, userInfo authenticationv1.UserInfo, v *NetworkPolicyValidator) (string, bool)
}
```

similar to https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook/admission#Validator

Reply:
updated